src/common/spawn.cpp

```diff
@@ -29,16 +29,68 @@
 #include <errno.h>
 #include <unistd.h>
 #include <sys/types.h>
+#ifdef _WIN32
+#include <boost/algorithm/string/join.hpp>
+#include <boost/scope_exit.hpp>
+#include <windows.h>
+#else
 #include <sys/wait.h>
+#endif
 
 #include "misc_log_ex.h"
-#include "exec.h"
+#include "spawn.h"
 
 namespace tools
 {
 
-int exec(const char *filename, char * const argv[], bool wait)
+int spawn(const char *filename, const std::vector<std::string>& args, bool wait)
 {
+#ifdef _WIN32
+  std::string joined = boost::algorithm::join(args, " ");
+  char *commandLine = !joined.empty() ? &joined[0] : nullptr;
+  STARTUPINFOA si = {};
+  si.cb = sizeof(si);
+  PROCESS_INFORMATION pi;
+  if (!CreateProcessA(filename, commandLine, nullptr, nullptr, false, 0, nullptr, nullptr, &si, &pi))
```

Comment:
I suspect this should be using CreateProcessW and handling UTF8->UCS2 conversion. I.e., using CreateProcessA will probably break if there are non-ASCII filenames in the path.

Reply:
Looking into this. Have to check if we actually have utf8 strings here (`const char *filename, const std::vector<std::string>& args`)

Reply:
Inspected the case. Is the valid point, but it covers very rare use case and requires performing major update of other code parts non-relating to the current PR.

Current code will work just fine with the paths containing local system CP symbols. This will suit most users' needs. 

A Monero user willing to set a notification containing non-local CP characters looks like an extra rare case. I wonder how he will type these characters in the console.

Handling this requires implementing Unicode command line support.
Current command line code parses ANSI arguments (`char const * argv[]`) that are encoded in local system code page on Windows.

Reply:
Fair enough.